src/rgw/rgw_data_sync.cc

```diff
@@ -1182,9 +1182,12 @@ class RGWDataSyncCR : public RGWCoroutine {
   }
 
   ~RGWDataSyncCR() {
+    shard_crs_lock.Lock();
```

Comment:
@cbodley you have a point, I've seen shard_crs inconsistency, but can't be sure what the culprit was.